src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -2198,7 +2199,21 @@ def serialize_key_and_certificates_to_pkcs12(
             # This list is to keep the x509 values alive until end of function
             ossl_cas = []
             for ca in cas:
-                ossl_ca = self._cert2ossl(ca)
+                if isinstance(ca, PKCS12Certificate):
+                    # Copied from Felix's gist
+                    # https://gist.github.com/
+                    # felixfontein/f750763fd5773fbf0ab22af87239aab0
+                    ca_alias = ca.friendly_name
+                    ossl_ca = self._cert2ossl(ca.certificate)
+                    if ca_alias is not None:
+                        with self._zeroed_null_terminated_buf(
+                            ca_alias
+                        ) as ca_name_buf:
+                            self._lib.X509_alias_set1(
```

Comment:
Check the result of this function.

Reply:
It now checks the result and throws a `RuntimeError`. However, I think that due to the checks in the method it's impossible to get to the fail state so coverage might complain.

If I read the source correctly, `X509_alias_set1` can only fail if:
- `x509 == self._ffi.NULL`.
- If the `aux` member of `x509` is `NULL` and cannot be created.
- If the `aux->alias` field of `x509` is 'NULL` and cannot be instantiated.
- If the length of `ca_name_buf` is very long.
- If the memory allocation of the alias string fails deep at the C-level.

I have no idea how to trigger these conditions by calling `serialize_key_and_certificates_to_pkcs12`, the checks it makes seem to guard against `X509_alias_set1` failure. Can you think of a way to make it fail?

Reply:
Also, I removed a the `if ca_alias is not None` check. `X509_alias_set1` can take arbitrary length null-terminated strings as input if the last argument is -1. Less nesting is better.

Reply:
You can use `self.openssl_assert()`, which is how we usually handle these "won't fail in practice" checks.

Reply:
Good to know, thanks!